### ISSUE TITLE
False positives about missing props for functional components

### ISSUE THREAD

**Issue by 0x80** [yannickcr/eslint-plugin-react#472]
With the current version the code below gives false positives about missing proptypes for all props.

```javascript
import { PropTypes } from 'react'
const { element, node, string } = PropTypes

Card.propTypes = {
  title: string.isRequired,
  children: element.isRequired,
  footer: node
}

export default function Card ({ title, children, footer }) {
  return (
    <div/>
  )
}

```

eslint 2.2.0
eslint-plugin-react 4.1.0
eslint-config-standard-react 2.3.0

.eslintrc
```json
{
  "parser": "babel-eslint",
  "extends": ["standard", "standard-react"],
  "rules" : {
    "react/react-in-jsx-scope": 0,
    "padded-blocks" : 0,
    "space-before-function-paren": 0,
    "no-multi-spaces": 0,
    "key-spacing": 0,
    "arrow-parens": 0
  }
}
```

**Comment by smaye81** [comment]
I am still seeing false positives with the following syntax.  Not sure what I'm missing:

```
import React, { PropTypes } from 'react';
import {
    ProgressBar
} from 'react-bootstrap';

const TableProgressBar = React.createClass({

    render () {
        return (
            <tr className="table-progress-bar">
                <td colSpan="5">
                    <ProgressBar active now={100} />
                    <p>{this.props.message}</p>
                </td>
            </tr>
        )
    }
});

TableProgressBar.propTypes = {
    message: PropTypes.string
};

export default TableProgressBar
```

**Comment by smaye81** [comment]
So it looks like this might not be supported syntax according to this:

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md

Looks like the `Component.propTypes = ...` is only supported with stateless functional components, but not with `React.createClass`.  I sorta feel like it should be though.  Thoughts?  If so, I can create a PR

**Comment by ljharb** [comment]
It definitely shouldn't be because I'm pretty sure that doesn't work. With React.createClass, you need to include `propTypes` in the actual object passed to `React.createClass`.

**Comment by smaye81** [comment]
No, you don't have to do it that way.  My entire app is structured like my example above and it works fine.  

I think this prop-types rule is a great rule and I want to be able to apply it without restructuring every component in my application.

**Comment by ljharb** [comment]
That's entirely undocumented and discouraged (since `createClass` is deprecated anyways) - I'm not sure it's a good thing to support.

**Comment by smaye81** [comment]
`React.createClass` is not deprecated nor can I find anywhere that says this syntax is discouraged. But no worries. I'll refactor to adhere to the supported syntax.
